manifests/iop_advisor_engine.pp

```diff
@@ -0,0 +1,69 @@
+# Contains certs specific configurations for advisor
+class certs::iop_advisor_engine (
+  Stdlib::Fqdn $hostname = 'localhost',
+  Boolean $generate = $certs::generate,
+  Boolean $regenerate = $certs::regenerate,
+  Boolean $deploy = $certs::deploy,
+  String[2,2] $country = $certs::country,
+  String $state = $certs::state,
+  String $city = $certs::city,
+  String $org = $certs::org,
+  String $org_unit = $certs::org_unit,
+  String $expiration = $certs::expiration,
+  Stdlib::Absolutepath $ca_key_password_file = $certs::ca_key_password_file,
+  String $owner = 'root',
+  String $group = 'root',
+  Stdlib::Filemode $private_key_mode = '0440',
+  Stdlib::Filemode $public_key_mode = '0444',
+) inherits certs {
+  include certs::foreman_proxy
+
+  $server_cert_name = "${hostname}-iop-advisor-server"
+  $server_cert = "${cert_directory}/server.cert"
+  $server_key = "${cert_directory}/server.key"
+  $server_ca_cert = $certs::katello_server_ca_cert
+
+  $client_cert = $certs::foreman_proxy::foreman_ssl_cert
+  $client_key = $certs::foreman_proxy::foreman_ssl_key
+  $client_ca_cert = $certs::foreman_proxy::foreman_ssl_ca_cert
+
+  cert { $server_cert_name:
+    ensure        => present,
+    hostname      => $hostname,
+    country       => $country,
+    state         => $state,
+    city          => $city,
+    org           => $org,
+    org_unit      => $org_unit,
+    expiration    => $expiration,
+    ca            => $certs::default_ca,
+    generate      => $generate,
+    regenerate    => $regenerate,
+    password_file => $ca_key_password_file,
+    build_dir     => $certs::ssl_build_dir,
+  }
+
+  if $deploy {
+    $cert_directory = '/etc/iop-advisor-engine'
```

Comment:
line 22/23 use that, so I guess *this* should be  outside the if?

Reply:
Does it?

![idk-look-harder](https://github.com/user-attachments/assets/7198d99d-6191-4cf1-8962-049181c637bd)